shania_cox/alpha_challenges/string_manipulation.md

```diff
@@ -0,0 +1,58 @@
+# String Manipulation
+
+## Allen Iverson
+
+In 2002, Allen Iverson told reporters how he felt about practice, famously using the word 'practice' over 20 times. Let's use code to pick apart his rant. It's pretty long. Wouldn't it be nice if we could ***split*** his rant up by individual words and ***iterate*** through each word to manipulate his rant? That'd be sweet. Then maybe we could ***join*** his rant back together afterwards, or something.
+
+<img src="http://ballertainment.com/wp-content/uploads/2012/12/Allen_Iverson_Practice.jpg">
+
+```ruby
+iverson_rant = "If I can't practice, I can't practice, man. If I'm hurt, I'm hurt. I mean simple as that. It ain't about that. I mean it's, it's not about that at all. You know what I'm saying. I mean, but it's, it's easy to, to talk about. It's easy to sum it up when you're just talking about practice. We're sitting in here, and I'm supposed to be the franchise player, and we in here talking about practice. I mean, listen, we're talking about practice, not a game, not a game, not a game, we talking about practice. Not a game. Not, not. Not the game that I go out there and die for and play every game like it's my last. Not the game, but we're talking about practice, man. I mean, how silly is that? And we talking about practice. I know I'm supposed to be there. I know I'm supposed to lead by example. I know that, and I'm not, I'm not shoving it aside, you know, like it don't mean anything. I know it's important, I do. I honestly do. But we're talking about practice, man. What are we talking about? We're talking about practice, man."
+```
+
+1. Iverson really ranted. We want to create an acronym for his amazing rant. It would look something like this.
+
+```ruby
+"IIcpIcpmIIhIhImsatIaatImiInatAaYkwIsImBiietttaIetsiuwyjtapWsihaIstbtfpawihtapImlwtapnagnagnagwtapNagNnNtgtIgotadfapeglimlNtgbwtapmImhsitAwtapIkIstbtIkIstlbeIktAinInsiayklidmaIkiiIdIhdBwtapmWawtaWtapm"
+```
+
+Programmatically manipulate `iverson_rant` to reflect the output above.
+
+```ruby
+words = iverson_rant.split(" ")
+words.each do |word|
+  print word[0]
```

Comment:
This is one way to approach it, but I don't think you're getting the exact output you want. `print`ing does print the acronym, but the original `words` array is what is returned (plus a weird little modified bonus):
```
IIcpIcpmIIhIhImsatIaatImiinataaYkwIsImbiietttaIetsiuwyjtapWsihaIstbtfpawihtapImlwtapnagnagnagwtapNagNnNtgtIgotadfapegli
=> ["If",pmImhsitAwtapIkIstbtIkIstlbeIktaInInsiayklidmaIkiiIdIhdBwtapmWawtaWtapm
 "I",
 "can't",
 "practice,",
 "I",
 "can't",
 "practice,",
 "man.",
 "If",
 "I'm",
 "hurt,", etc.
```
How would you return _just_ that acronym? Here's a hint: look into `.map` and `.join`.

Reply:
ruby has a nify `#chr` method, which will return the first character of your string.